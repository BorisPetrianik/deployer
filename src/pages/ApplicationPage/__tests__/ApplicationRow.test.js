import React, { useState } from 'react'
import { makeStyles } from '@material-ui/core/styles'
// import { TableCell, TableRow, IconButton, Tooltip, CircularProgress, Grid } from '@material-ui/core'
// import { IApplicationInstance } from 'types/Application'
import PlayArrowIcon from '@material-ui/icons/PlayArrow'
import StopIcon from '@material-ui/icons/Stop'
import ReplayIcon from '@material-ui/icons/Replay'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
import { ApplicationInstanceRow } from '../InstancesTable/ApplicetionInstanceRow'
import { TableCell, TableRow, IconButton, Tooltip, CircularProgress, Grid } from '@material-ui/core'
Enzyme.configure({ adapter: new Adapter() });



describe('ApplicationInstanceRow', ()=>{
    let data = {status: 'RUNNING'}
    beforeEach(()=> {
        data = {status: 'RUNNING'}
    })
    it('renders correctly', () => {
        const CheckRow = renderer
          .create(<ApplicationInstanceRow data={data}/>)
          .toJSON();
        expect(CheckRow).toMatchSnapshot();
      });

      it('Should contain at least 4 sells', () => {
        const wrapper = shallow(<ApplicationInstanceRow data={data}/>);
        expect(wrapper.find(TableCell).length).toBeGreaterThan(3);
      });

      it('Should contain at least 4 sells', () => {
        const wrapper = shallow(<ApplicationInstanceRow data={data}/>);
        expect(wrapper.find(TableCell).length).toBeLessThan(10);
      });

      // it('Should contain right hader', () => {
      //   const wrapper = shallow(<ApplicationInstanceRow data={data}/>);
      //   expect(wrapper.find('h3').text()).toEqual('ApplicationPageTabdata');
      // });
})