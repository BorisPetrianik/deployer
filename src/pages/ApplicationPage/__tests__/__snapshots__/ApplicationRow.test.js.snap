// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationInstanceRow renders correctly 1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  />
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  />
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  />
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    RUNNING
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    <div
      className="MuiGrid-root makeStyles-rowActions-1 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          aria-label="start"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary makeStyles-startButtonColor-2 Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="start instance"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 5v14l11-7z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          aria-label="stop"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary makeStyles-stopButtonColor-3 MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="stop instance"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          aria-label="restart"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="restart instance"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  />
</tr>
`;
